pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -62,7 +70,15 @@ func (a *actuator) Reconcile(ctx context.Context, bastion *extensionsv1alpha1.Ba
 		return err
 	}
 
-	nic, err := ensureNic(ctx, factory, opt, publicIP)
+	if infrastructureStatus.Networks.Layout != "SingleSubnet" {
```

Comment:
Does it really not work when there are multiple subnets? I would think that the bastion instance would be able to reach the other nodes.

Reply:
I didn't test it, the main Azure bastion  PR base on a single Subnet mode is completed before multiple subnets are achieved https://github.com/gardener/gardener-extension-provider-azure/pull/331. but we have an internal ticket about `Bastion Azure extension multi zones support`

Reply:
Hi @kon-angelo, there are Firewall rules created together with Bastion that limit connections to the certain subnet. Originally there was a plan to implement multi-AZ right after the main Bastion development but priorities were changed, unfortunately...

Reply:
Just in case I dismiss our internal ticket because it's not visible to anyone...

Reply:
after rethinking, I think this PR also covers multiple subnets supported by a bastion extension now. By default, Azure subnets communicate with each other under the same virtual network.  https://docs.microsoft.com/en-us/azure/virtual-network/virtual-networks-udr-overview#:~:text=Azure%20automatically%20routes%20traffic,of%20a%20virtual%20network

I am assuming `infrastructureStatus.Networks.Subnets` always keep subnets which used by the gardener only,  by default bastion uses the first subnet to create a bastion
https://github.com/gardener/gardener-extension-provider-azure/blob/5297c61f6db5db460b719a6094169dbb580a01bb/pkg/controller/bastion/actuator_reconcile.go#L81

will remove the `SingleSubnet` mode check logic

Reply:
done https://github.com/gardener/gardener-extension-provider-azure/pull/492/commits/042d16bd153e5af48243d0b339062a6a754c5569

Reply:
@tedteng I checked it and as it is just removing the `SingleSubnet` check is not enough.
Here is a solution in a draft PR i opened based on this one. You can do some testing yourself too to see that it works E2E.

Reply:
Thanks, It works. I will close the PR. 
Please process the [draft PR ](https://github.com/gardener/gardener-extension-provider-azure/pull/518/commits/b7081b62b2d8f5687918b9a9ac48550219e39497)you opened